core/templates/dev/head/domain/skill/ConceptCardBackendApiService.ts

```diff
@@ -48,69 +48,83 @@ oppia.factory('ConceptCardBackendApiService', [
-       * attempt to see whether the given concept card has already been loaded.
-       * If it has not yet been loaded, it will fetch the concept card from the
-       * backend. If it successfully retrieves the concept card from the
-       * backend, it will store it in the cache to avoid requests from the
-       * backend in further function calls.
+       * attempt to see whether the given concept cards have already been
+       * loaded. If they have not yet been loaded, it will fetch the concept
+       * cards from the backend. If it successfully retrieves the concept cards
+       * from the backend, it will store them in the cache to avoid requests
+       * from the backend in further function calls.
        */
-      loadConceptCard: function(skillId) {
+      loadConceptCards: function(skillIds) {
         return $q(function(resolve, reject) {
-          if (_isCached(skillId)) {
+          if (_isCached(skillIds)) {
             if (resolve) {
-              resolve(angular.copy(_conceptCardCache[skillId]));
+              var conceptCards = [];
+              skillIds.forEach(function(skillId) {
+                conceptCards.push(angular.copy(_conceptCardCache[skillId]));
+              });
+              resolve(conceptCards);
             }
           } else {
-            _fetchConceptCard(skillId, function(conceptCard) {
-              // Save the fetched conceptCard to avoid future fetches.
-              _conceptCardCache[skillId] = conceptCard;
+            _fetchConceptCards(skillIds, function(conceptCards) {
+              // Save the fetched conceptCards to avoid future fetches.
+              for (var i = 0; i < skillIds.length; i++) {
+                _conceptCardCache[skillIds[i]] = angular.copy(conceptCards[i]);
+              }
               if (resolve) {
-                resolve(angular.copy(conceptCard));
+                resolve(angular.copy(conceptCards));
               }
             }, reject);
           }
         });
       },
 
       /**
-       * Returns whether the given concept card is stored within the local data
-       * cache or if it needs to be retrieved from the backend upon a laod.
+       * Returns whether the given concept cards are stored within the local
+       * data cache or if they need to be retrieved from the backend upon a
+       * laod.
        */
-      isCached: function(skillId) {
-        return _isCached(skillId);
+      isCached: function(skillIds) {
+        return _isCached(skillIds);
       },
 
       /**
-       * Replaces the current concept card in the cache given by the specified
-       * skill ID with a new concept card object.
+       * Replaces the current concept cards in the cache given by the
+       * specified skill IDs with new concept card objects.
        */
-      cacheConceptCard: function(skillId, conceptCard) {
-        _conceptCardCache[skillId] = angular.copy(conceptCard);
+      cacheConceptCards: function(skillIds, conceptCards) {
```

Comment:
Is this function being used anywhere? I don't see it being used.

Reply:
It's not being used, but I think it's better to keep it there since it has existed before this PR.

Reply:
If its not being used let's remove it

Reply:
Yeah, it's not used anywhere. I don't think an external service/directive would want to cache new cards, as all new cards would be cached in the load function itself.

Reply:
Okay. I removed this function as well as `claerConceptCardCache` since I don't think it would be used outside of this service.